gulpfile.mjs

```diff
@@ -63,92 +56,19 @@ gulp.task(
-  const bootBundle = 'build/scripts/boot.bundle.js';
-
-  // In development, fail silently if the boot bundle has not yet been generated.
-  // This function will be re-run after it is. In production, the boot bundle
-  // should always have been generated first.
-  if (usingDevServer && !existsSync(bootBundle)) {
-    return;
-  }
-
-  const { version } = JSON.parse(readFileSync('./package.json').toString());
-
-  const defaultNotebookAppUrl = process.env.NOTEBOOK_APP_URL
-    ? `${process.env.NOTEBOOK_APP_URL}`
-    : '{current_scheme}://{current_host}:5000/notebook';
-
-  const defaultSidebarAppUrl = process.env.SIDEBAR_APP_URL
-    ? `${process.env.SIDEBAR_APP_URL}`
-    : '{current_scheme}://{current_host}:5000/app.html';
-
-  let defaultAssetRoot;
-
-  if (process.env.NODE_ENV === 'production' && !usingDevServer) {
-    defaultAssetRoot = 'https://cdn.hypothes.is/hypothesis';
-  } else {
-    defaultAssetRoot = '{current_scheme}://{current_host}:3001/hypothesis';
-  }
-  defaultAssetRoot = `${defaultAssetRoot}/${version}/`;
-
-  if (isFirstBuild) {
-    log(`Sidebar app URL: ${defaultSidebarAppUrl}`);
-    log(`Notebook app URL: ${defaultNotebookAppUrl}`);
-    log(`Client asset root URL: ${defaultAssetRoot}`);
-    isFirstBuild = false;
-  }
-
-  gulp
-    .src(bootBundle)
-    .pipe(replace('__MANIFEST__', JSON.stringify(manifest)))
-    .pipe(replace('__ASSET_ROOT__', defaultAssetRoot))
-    .pipe(replace('__NOTEBOOK_APP_URL__', defaultNotebookAppUrl))
-    .pipe(replace('__SIDEBAR_APP_URL__', defaultSidebarAppUrl))
-    // Strip sourcemap link. It will have been invalidated by the previous
-    // replacements and the bundle is so small that it isn't really valuable.
-    .pipe(replace(/^\/\/# sourceMappingURL=\S+$/m, ''))
-    .pipe(rename('boot.js'))
-    .pipe(gulp.dest('build/'));
-}
-
-/**
- * Generate a JSON manifest mapping file paths to
- * URLs containing cache-busting query string parameters.
- */
-function generateManifest(opts) {
-  return gulp
-    .src(MANIFEST_SOURCE_FILES)
-    .pipe(manifest({ name: 'manifest.json' }))
-    .pipe(
-      through.obj(function (file, enc, callback) {
-        const newManifest = JSON.parse(file.contents.toString());
-
-        // Expand template vars in boot script bundle
-        generateBootScript(newManifest, opts);
-
-        this.push(file);
-        callback();
-      })
-    )
-    .pipe(gulp.dest('build/'));
-}
-
-gulp.task('watch-manifest', () => {
-  gulp.watch(MANIFEST_SOURCE_FILES, { delay: 500 }, function updateManifest() {
-    return generateManifest({ usingDevServer: true });
-  });
+const manifestSourceFiles = 'build/{scripts,styles}/*.{css,js,map}';
+
+gulp.task('build-boot-script', async () => {
+  await generateManifest({ pattern: manifestSourceFiles });
+  await buildJS('./rollup-boot.config.mjs');
+});
+
+gulp.task('watch-boot-script', () => {
+  gulp.watch(
+    manifestSourceFiles,
+    { delay: 500 },
```

Comment:
This comes from pre-existing code, but curious why a half-second delay is needed here?

Reply:
IIRC this was done to prevent too many unnecessary re-executions of this task when a set of files were changed in quick succession.